pennylane/circuit_graph.py

```diff
@@ -588,10 +592,27 @@ def greedy_layers(self):
             if not observables[wire]:
                 observables[wire] = [None]
 
-        return (
-            [operations[wire] for wire in operations],
-            [observables[wire] for wire in observables],
-        )
+        if wire_order is not None:
```

Comment:
It is generally safer to always compare singletons directly using `is` or `is not`, rather than relying on boolean logic! This is recommended by both [PEP8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) and Google's style guide:

> Comparisons to singletons like None should always be done with is or is not, never the equality operators.
>
> Also, beware of writing if x when you really mean if x is not None -- e.g. when testing whether a variable or argument that defaults to None was set to some other value. The other value might have a type (such as a container) that could be false in a boolean context!

The reason being is that `if x is not None:` is more general than `if x:`. For example, `x=""` will evaluate as `True` for the first if statement, as we require, but `False` for the second.

Reply:
Good to know.  Learn new things every day :)

Is there ever a time when `if x` is good to use?

Reply:
Yep! You can use `if x` if you know that `bool(x)` evaluate to true only when the branch is activated.

E.g., if you want to branch only if a string is non-empty, you can do `if str_variable:`. Since `bool("")` evaluates to False, but `bool("non empty string")` evaluates to True.

Basically:

* Use `if x:` if your condition is an *equality* (it depends on the _value_ of the object)
* Use `if x is ___` if your condition is an *identity* (it depends on a specific object)